recipes/flann/all/conanfile.py

```diff
@@ -0,0 +1,117 @@
+import glob
+import os
+
+from conans import ConanFile, CMake, tools
+
+
+class LibFlannConan(ConanFile):
+    name = "flann"
+    description = "Fast Library for Approximate Nearest Neighbors"
+    topics = "conan", "flann"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://www.cs.ubc.ca/research/flann/"
+    license = "BSD-3-Clause"
+    exports_sources = "CMakeLists.txt"
+    generators = "cmake", "cmake_find_package"
+
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "with_hdf5": [True, False]
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "with_hdf5": False
+    }
+
+    _source_subfolder = "source_subfolder"
+    _build_subfolder = "build_subfolder"
+    _cmake = None
+
+    def config_options(self):
+        if self.settings.compiler == "Visual Studio":
+            del self.options.fPIC
+
+    def requirements(self):
+        if self.options.with_hdf5:
+            self.requires("hdf5/1.10.6")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = self.name + "-" + self.version
+        os.rename(extracted_dir, self._source_subfolder)
+
+        # Workaround issue with empty sources for a CMake target
+        flann_cpp_dir = os.path.join(self._source_subfolder, "src", "cpp")
+        tools.save(os.path.join(flann_cpp_dir, "empty.cpp"), "\n")
+
+        tools.replace_in_file(
+            os.path.join(flann_cpp_dir, "CMakeLists.txt"),
+            'add_library(flann_cpp SHARED "")',
+            'add_library(flann_cpp SHARED empty.cpp)'
+        )
+        tools.replace_in_file(
+            os.path.join(flann_cpp_dir, "CMakeLists.txt"),
+            'add_library(flann SHARED "")',
+            'add_library(flann SHARED empty.cpp)'
+        )
+
+    def _configure_cmake(self):
+        if self._cmake is not None:
+            return self._cmake
+        self._cmake = CMake(self)
+
+        # Only build the C++ libraries
+        self._cmake.definitions["BUILD_DOC"] = "OFF"
```

Comment:
for the record, you may use `False`, but it's just a minor thing (not blocking)

Reply:
I often wondered which of the two was more idiomatic.

Reply:
I prefer **False**, because it could be validated, while string constant could be easily misspelled and you will never notice (e.g. may use zero instead of O, or Cyrillic O instead of latic, etc.). but it's not so important anyway, just for the future :)

Reply:
I still have a small linkable example to add to the recipe, so I'll probably do that as a drive-by fix. Thanks :)

Reply:
I changed the options to only use booleans instead of strings.